app/helpers/html_helper.php

```diff
@@ -5,3 +5,62 @@ function eh($string)
     if (!isset($string)) return;
     echo htmlspecialchars($string, ENT_QUOTES);
 }
+
+function readable_text($s) {
+    $s = htmlspecialchars($s, ENT_QUOTES);
+    $s = nl2br($s);
+    return $s;
+}
+
+// trim whitespace from $str and collapse adjacent whitespaces into one.
+function trim_collapse($str) {
+    $str = trim($str);
+    $str = preg_replace('/[[:space:]]{2,}/', ' ', $str);
+    return $str;
+}
+
+function redirect($str) {
+    header("Location: " . url($str));
+    die();
```

Comment:
Not when the header() fails po. I read that every time you set the location header, die() or exit() should be called to stop script execution. 
http://stackoverflow.com/questions/8665985/php-utilizing-exit-or-die-after-headerlocation